app/controllers/application_controller.rb

```diff
@@ -5,6 +5,67 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions unless test?
+    set :session_secret, "secret"
   end
 
+  get '/' do
+  erb :index
+  end
+
+  get '/tweets/new' do
+    erb :'tweets/create_tweet'
+  end
+
+  post '/tweets' do
+    tweet = Tweet.create(params)
+    tweet.save
+    redirect('/tweets')
+  end
+
+  get '/tweets' do
+    @tweets = Tweet.all
+    erb :'tweets/tweets'
+  end
+
+  get '/signup' do
+    if !session
+     erb :'users/create_user'
+    else
+    redirect('/tweets')
+    end
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.